crowbar_framework/app/controllers/installer/upgrades_controller.rb

```diff
@@ -23,6 +23,27 @@ class UpgradesController < ApplicationController
     before_filter :set_progess_values
     before_filter :set_service_object, only: [:services, :backup, :nodes]
 
+    def prepare
+      status = :ok
+
+      if request.post?
+        begin
+          service_object = CrowbarService.new(Rails.logger)
+
+          service_object.prepare_nodes_for_crowbar_upgrade
+        rescue => e
+          Rails.logger.error e.message
+          status = :unprocessable_entity
+        end
+      end
```

Comment:
So a get would fail due to missing route - ok fine with me then.
I guess I would keep the `if request.post?` even though.